plumbing/transport/ssh/common.go

```diff
@@ -11,7 +11,16 @@ import (
 )
 
 // DefaultClient is the default SSH client.
-var DefaultClient = common.NewClient(&runner{})
+var DefaultClient = NewClient()
+
+// NewClient creates a new SSH client with the given options.
+func NewClient(opts ...ClientOption) transport.Transport {
```

Comment:
To play nice with other options that modify `ssh.ClientConfig`:

```
 	config := c.auth.clientConfig()
 	config.HostKeyCallback, err = c.auth.hostKeyCallback()
 	if err != nil {
 		return err
 	}
```

`ssh.ClientConfig` is created and initialized by client on each session. Some fields are set based on `AuthMethod`, which is not global, but per-session setting. Then setting the new options for `ssh.ClientConfig` are set globally for the transport, but applied per session after it has been created.